e2e/cypress/integration/ad_ldap/saml_ldap_sync_id_attrib_spec.js

```diff
@@ -0,0 +1,185 @@
+    const admin = getAdminAccount();
+    const baseUrl = Cypress.config('baseUrl');
+    const loginButtonText = 'SAML';
+
+    const user3 = {
+        username: 'e2etest.three.saml',
+        password: 'Password1',
+        email: 'e2etest.three@mmtest.com',
+        firstname: 'FirstSaml',
+        lastname: 'ThreeSaml',
+        ldapfirstname: 'TestLDAP',
+        ldaplastname: 'ThreeLDAP',
+        keycloakId: '',
+    };
+    const users = {user3};
+
+    const {
+        keycloakBaseUrl,
+        keycloakAppName,
+    } = Cypress.env();
+
+    const idpUrl = `${keycloakBaseUrl}/auth/realms/${keycloakAppName}/protocol/saml`;
+    const idpDescriptorUrl = `${keycloakBaseUrl}/auth/realms/${keycloakAppName}`;
+
+    const newConfig = {
+        SamlSettings: {
+            Enable: true,
+            EnableSyncWithLdap: false,
+            EnableSyncWithLdapIncludeAuth: false,
+            Verify: true,
+            Encrypt: false,
+            SignRequest: false,
+            IdpUrl: idpUrl,
+            IdpDescriptorUrl: idpDescriptorUrl,
+            IdpMetadataUrl: '',
+            ServiceProviderIdentifier: `${Cypress.config('baseUrl')}/login/sso/saml`,
+            AssertionConsumerServiceURL: `${Cypress.config('baseUrl')}/login/sso/saml`,
+            SignatureAlgorithm: 'RSAwithSHA256',
+            CanonicalAlgorithm: 'Canonical1.0',
+            IdpCertificateFile: 'saml-idp.crt',
+            PublicCertificateFile: '',
+            PrivateKeyFile: '',
+            IdAttribute: 'username',
+            GuestAttribute: '',
+            EnableAdminAttribute: false,
+            AdminAttribute: '',
+            FirstNameAttribute: 'firstName',
+            LastNameAttribute: 'lastName',
+            EmailAttribute: 'email',
+            UsernameAttribute: 'username',
+            LoginButtonText: loginButtonText,
+        },
+        LdapSettings: {
+            EnableSync: true,
+            BaseDN: 'ou=e2etest,dc=mm,dc=test,dc=com',
+        },
+    };
+
+    let testSettings;
+
+    before(() => {
+        // * Check if server has license for SAML
+        cy.apiRequireLicenseForFeature('SAML');
+
+        // * Check if server has license for LDAP
+        cy.apiRequireLicenseForFeature('LDAP');
+
+        // # Get certificates status and upload as necessary
+        cy.apiGetSAMLCertificateStatus().then((resp) => {
+            const data = resp.body;
+
+            if (!data.idp_certificate_file) {
+                cy.apiUploadSAMLIDPCert('saml-idp.crt');
+            }
+        });
+
+        // # Update Configs
+        cy.apiUpdateConfig(newConfig).then(({config}) => {
+            cy.setTestSettings(loginButtonText, config).then((_response) => {
+                testSettings = _response;
+                cy.keycloakResetUsers(users);
+            });
+        });
+
+        // # Add/refresh LDAP Test users
+        cy.addLDAPUsers();
+    });
+
+    it('MM-T3013 (Step 4) - SAML / LDAP sync with ID Attribute', () => {
```

Comment:
Our format for test case with multiple steps is `[tm4j_id]_[step_number] [title]` so that we can exactly match the steps when saving/integrating test execution to Zephyr/TM4J. Therefore this should be 
```javascript
    it('MM-T3013_4 SAML / LDAP sync with ID Attribute', () => {
```

Btw, since each step is huge and can be tested in isolation, I'm leaning towards extracting each step into its own test case. If you agree, I'll update the Zephyr/TM4J.

Reply:
I updated to match steps. I'm ok with extracting into its own test case. Let me know.

Reply:
Thanks! Done. Made comments to `it` block itself.